docs/use_cases/retrieval_augmented_generation_eval.md

```diff
@@ -0,0 +1,181 @@
+tools. The tools that you use should cover each component of RAG. Such coverage
+ensures granular and thorough measurements.
+
+Evaluation metrics can assess:
+
+- Retrieval effectiveness
+- Coherence of generated responses
+- Relevance to the retrieved information.
+
+<img src=/assets/use_cases/retrieval_augmented_generation_eval/rag_granular.jpg alt="Granular Levels of Evaluation of RAG" data-size="100" />
+
+### Let's dive in !!
+
+## Strategies for Evaluation
+
+Now that we have defined different challenges and levels at which we can break down the RAG evaluation, let us zoom into the levels individually. 
+
+### Model Evaluation 
+
+To evaluate our model, we start with this [Massive Text Embedding Benchmark](https://huggingface.co/spaces/mteb/leaderboard). 
+We want to ensure that the model can understand the data that we encode.
+The benchmark shown above leverages different public/private datasets to evaluate and report on the different capabilities of individual models.
+If you're working with specialized domains, you may want to put together a specialized dataset to teach the model.
+
+We could either leverage the results here (provided our model belongs to this list) or run relevant 'tasks' for our custom model (instructions provided in the github [link](https://github.com/embeddings-benchmark/mteb#leaderboard))
+
+```python
+import logging
+
+from mteb import MTEB
+from sentence_transformers import SentenceTransformer
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+model_name = "average_word_embeddings_komninos"
+model = SentenceTransformer(model_name)
+evaluation = MTEB(task_langs=["en"])
+evaluation.run(model, output_folder=f"results/{model_name}", eval_splits=["test"])
+
+print("--DONE--")
+```
+
+### Data Ingestion Evaluation
+
+We first validate the model, and train it on the language of our domain. 
+We can then configure data ingestion into our semantic retrieval store aka vector store.
+Various vector databases offer index configurations to influence and enhance the retrieval quality, based on the supported index types (Flat , LSH , HNSW and IVF).
+One such example here to improve HNSW [retrieval-quality](https://qdrant.tech/documentation/tutorials/retrieval-quality/).
+
+To evaluate ingestion, we need to focus on related variables, such as:
+
+* **Chunk size** - represents the size of each segment. This depends on the token limit of our embedding model. It also has a substantial impact on the contextual understanding of the data. That impacts the precision, recall, and relevancy of our results.
+
+* **Chunk overlap** - represents the presence of overlapping information fragments in each segment. This helps with context retention of the information chunks but at the same time must be used with relevant strategies like deduplication, and content normalization to eradicate adverse effects.
+
+* **Chunking/Text splitting strategy** - represents the process of data splitting and further treatment as mentioned above.
```

Comment:
how is this different from chunk_size and chunk_overlap? I suppose you mean what constitute the base splits that are later merged into chunks, right?

Reply:
This was the basic idea behind it : https://js.langchain.com/docs/modules/data_connection/document_transformers/#types-of-text-splitters

Do you suggest I explicitly add a couple of examples to facilitate understanding?

Reply:
I think I would make it more clear, that there are splits, which are merged to constitute chunks. And there are considerations when creating splits, and also when merging splits to arrive at chunks.